src_sycl/CMakeLists.txt

```diff
@@ -21,14 +21,43 @@ add_executable(nbodygl ${COMMON_SOURCE} ${OPENGL_SOURCE})
 
 target_compile_definitions(nbodygl PRIVATE -DUSE_OPENGL)
 
-target_compile_features(nbodygl PRIVATE cxx_auto_type cxx_nullptr cxx_range_for)
+target_compile_features(nbodygl PRIVATE cxx_auto_type cxx_nullptr cxx_range_for cxx_std_17)
 
 target_include_directories(nbodygl PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 target_link_libraries(nbodygl PRIVATE glm::glm glfw PkgConfig::Glew OpenGL::OpenGL)
 
-if(CMAKE_CXX_COMPILER MATCHES ".*dpcpp[.exe]*")
-	target_compile_options(nbodygl PRIVATE -fsycl)
-endif()
+option(USE_COMPUTECPP "whether to use ComputeCpp compiler" OFF)
+
+if(USE_COMPUTECPP)
+	find_package(ComputeCpp REQUIRED)
+
+	set(CMAKE_CXX_COMPILER ${ComputeCpp_DEVICE_COMPILER_EXECUTABLE})
```

Comment:
This isn't necessary, but is possible. If you do do it, then it sets `-sycl-driver` mode automatically in `FindComputeCpp.cmake`.